analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocations.cs

```diff
@@ -136,11 +137,73 @@ End Sub
         }
 
         [DataTestMethod]
-        [DataRow("DoSomething();")]
-        [DataRow("this.DoSomething();")]
-        [DataRow("(this).DoSomething();")]
-        [DataRow("(((this))).DoSomething();")]
-        public void Invocation_InstanceMethodCallDoesNotClearFieldOnThis(string invocation)
+        [DataRow("Initialize();")]
+        [DataRow("this.Initialize();")]
+        [DataRow("(this).Initialize();")]
+        [DataRow("(((this))).Initialize();")]
+        [DataRow("((IDisposable)this).Dispose();")]
+        [DataRow("((IDisposable)(object)this).Dispose();")]
+        [DataRow("this.SomeExtensionOnSample();")]
+        [DataRow("Extensions.SomeExtensionOnSample(this);")]
+        [DataRow("this.SomeExtensionOnObject();")]
+        [DataRow("Extensions.SomeExtensionOnObject(this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((IDisposable)this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((object)(IDisposable)this);")]
+        [DataRow("((object)(IDisposable)this).SomeExtensionOnObject();")]
+        public void Invocation_InstanceMethodCallDoesClearFieldOnThis(string invocation)
+        {
+            var code = $@"
+using System;
+namespace N
+{{
+    public class Sample: IDisposable
+    {{
+        object field1 = null;
+        object field2 = null;
+        static object staticField1 = null;
+        static object staticField2 = null;
+
+        void CallToMethodsShouldResetFieldConstraints()
+        {{
+            field1 = null;
+            field2 = null;
+            staticField1 = null;
+            staticField2 = null;
+            {invocation}
+            Tag(""Field1"", field1);
+            Tag(""Field2"", field2);
```

Comment:
I don't see any difference between `1` and `2`. Let's remove them
```suggestion
            Tag(""InstanceField"", instanceField);
```

Reply:
It's there to ensure all fields are processed.

Reply:
So we can have one static and one instance for that purose